### PR TITLE
UI/fix stacked charts tooltip

### PR DESCRIPTION
**Pull request by hellobontempo**
Updates charts to use lineal and subsequently fix the tooltip not rendering
Screenshot 2024-04-22 at 10 57 23 AM
Screenshot 2024-04-24 at 11 29 47 AM

### CODE REVIEW

File: ui/app/components/clients/charts/line.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export default class LineChart extends Component<Args> {
   };
   // These functions are used by the tooltip
   formatCount = (count: number) => {
-    return formatNumbers([count]);
+    return numericalAxisLabel([count]);
```
Comment (hashishaw) [inline ui/app/components/clients/charts/line.ts:135]: Why are we sending this number as an array? Looks like `numericalAxisLabel` takes a number

File: ui/app/components/clients/charts/line.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 
 import Component from '@glimmer/component';
-import { SVG_DIMENSIONS, formatNumbers } from 'vault/utils/chart-helpers';
+import { SVG_DIMENSIONS, numericalAxisLabel } from 'vault/utils/chart-helpers';
```
Comment (hellobontempo) [inline ui/app/components/clients/charts/line.ts:7]: renamed to differentiate between `formatNumber` the template helper and the chart axis formatter

File: ui/app/components/clients/charts/vertical-bar-basic.hbs [hunk 3]
```diff
@@ -11,7 +11,7 @@
       (scale-linear range=(array 0 this.chartHeight) domain=this.yDomain)
       as |xScale yScale hScale|
     }}
-      <svg width={{width}} height={{this.chartHeight}} data-test-sync-bar-chart>
+      <svg width={{width}} height={{this.chartHeight}}>
```
Comment (hellobontempo) [inline ui/app/components/clients/charts/vertical-bar-basic.hbs:14]: this chart is no longer sync specific

File: ui/app/components/clients/charts/vertical-bar-stacked.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  label(legendKey: string) {
+    return this.args.chartLegend.find((l: Legend) => l.key === legendKey).label;
+  }
+
+  get chartData() {
+    let dataset: [string, number | undefined, string, KeyDataItem[]][] = [];
+    // each datum needs to be its own object
+    for (const key of this.dataKeys) {
+      const keyData: KeyDataItem[] = this.args.data.map((d: MonthlyChartData) => ({
+        timestamp: d.timestamp,
```
Comment (hellobontempo) [inline ui/app/components/clients/charts/vertical-bar-stacked.ts:76]: At the moment we don't have any charts that aren't monthly charts, so making this key dynamic felt like over-complicating this work. Though this was implemented with abstraction in mind so adding getters for the xKey will be fairly straightforward.

File: ui/app/styles/components/chart-container.scss
Comment (hellobontempo) [inline ui/app/styles/components/chart-container.scss:1]: moved from charts.scss

File: ui/app/styles/core/charts.scss [hunk 5]
```diff
@@ -228,6 +38,7 @@ p.data-details {
   border-radius: $radius-large;
   flex-wrap: nowrap;
   width: fit-content;
+  margin-bottom: $spacing-8;
```
Comment (hellobontempo) [inline ui/app/styles/core/charts.scss:41]: ## before
Screenshot 2024-04-24 at 12 23 38 PM

## after
Screenshot 2024-04-24 at 12 22 25 PM

File: ui/tests/helpers/clients/client-count-selectors.ts [hunk 6]
```diff
@@ -68,11 +68,12 @@ export const CLIENT_COUNT = {
   upgradeWarning: '[data-test-clients-upgrade-warning]',
 };
 
-export const CHART_ELEMENTS = {
-  entityClientDataBars: '[data-test-group="entity_clients"]',
```
Comment (hellobontempo) [inline ui/tests/helpers/clients/client-count-selectors.ts:72]: these weren't used anywhere, plan is to migrate nested selectors above ☝️  into this block

**Review by hashishaw — APPROVED** [review]
Beautiful work, thanks for all the comments!

**Pull request merged**